docs/esp32/tutorial/reset.rst

```diff
@@ -0,0 +1,25 @@
+Factory reset
+=============
+
+If something unexpected happens and your ESP32-based board no longer boots
+MicroPython, then you may have to factory reset it. For more details, see
+:ref:`soft_bricking`.
+
+Factory resetting the MicroPython esp32 port involves fully erasing the flash
+and resetting the flash memory, so you will need to re-flash the MicroPython
+firmware afterwards and copy any Python files to the filesystem again.
+
+1. You will need the Espressif `esptool`_ installed on your system. This is the
+   same tool that you may have used to initially install MicroPython on your
+   board (see :ref:`installation instructions <esp32_flashing>`).
+2. Find the serial port name of your board, and then use esptool to erase the
+   entire flash contents::
+
+       esptool.py -p PORTNAME erase_flash
+
+3. Use esptool to flash the MicroPython firmware to your board again. If needed,
+   this file and flashing instructions can be found on the `MicroPython
```

Comment:
file -> firmware

Reply:
Have changed, but would like to note that this term was adopted from the linked esp32 tutorial which uses it extensively, for example "And then deploy the new firmware using::". (I did change "deploy" to "flash", though.)

(Can change the tutorial as well, but suggest I do it in the PR mentioned above.)